tools/parse/src/main/scala/org/allenai/nlpstack/parse/poly/polyparser/ArcHybridTransitionSystem.scala

```diff
@@ -366,13 +370,48 @@ case class ArcHybridForbiddenArcLabelInterpretation(
       case LabelLeftArc(arcLabel) =>
         (PreviousLinkCrumbRef(state).headOption, PreviousLinkGretelRef(state).headOption) match {
           case (Some(crumb), Some(gretel)) =>
-            Set(crumb, gretel) == arcTokens && arcLabel == forbiddenArcLabel.arcLabel
+            Set(crumb, gretel) == arcTokens &&
+              (DependencyParsingTransitionSystem.getArcLabelSymbol(arcLabel) ==
+                forbiddenArcLabel.arcLabel)
           case _ => false
         }
       case LabelRightArc(arcLabel) =>
         (PreviousLinkCrumbRef(state).headOption, PreviousLinkGretelRef(state).headOption) match {
           case (Some(crumb), Some(gretel)) =>
-            Set(crumb, gretel) == arcTokens && arcLabel == forbiddenArcLabel.arcLabel
+            Set(crumb, gretel) == arcTokens &&
+              (DependencyParsingTransitionSystem.getArcLabelSymbol(arcLabel) ==
+                forbiddenArcLabel.arcLabel)
+          case _ => false
+        }
+      case _ => false
+    }
+  }
+}
+
+/** The ArcHybridRequestedCposInterpretation handles RequestedCpos constraints for the
+  * arc hybrid system. In other words, it translates these constraints into a function that
+  * returns true for any (state, transition) pair that violates the constraint.
+  *
+  * @param requestedCpos coarse POS request to consider
+  */
+case class ArcHybridRequestedCposInterpretation(
+    requestedCpos: RequestedCpos
+) extends ParsingConstraintInterpretation {
+
+  def applyToParserState(state: TransitionParserState, transition: StateTransition): Boolean = {
+    transition match {
+      case LabelLeftArc(arcLabel) =>
+        PreviousLinkGretelRef(state).headOption match {
+          case Some(gretel) =>
+            gretel == requestedCpos.tokenIndex &&
+              (DependencyParsingTransitionSystem.getArcLabelCpos(arcLabel) != requestedCpos.cpos)
+          case _ => false
+        }
+      case LabelRightArc(arcLabel) =>
+        PreviousLinkGretelRef(state).headOption match {
+          case Some(gretel) =>
+            gretel == requestedCpos.tokenIndex &&
+              (DependencyParsingTransitionSystem.getArcLabelCpos(arcLabel) != requestedCpos.cpos)
```

Comment:
Again I don't think you need wrapping parens here or above (but it's not a big deal)